### ISSUE TITLE
No unused variables for stateless functional components

### ISSUE THREAD

**Issue by 0x80** [feross/eslint-config-standard-react#21]
I would expect the code below to warn me about the fact that `bar` is not used. 

I was trying out [xo](https://github.com/sindresorhus/xo) the other day and it does give that warning. It is also based on eslint-plugin-react, so I'm wonder if it's just a difference in settings...

``` javascript
import {PropTypes} from 'react'

Foo.propTypes = {bla: PropTypes.string}

export default function Foo ({bar}) {
  return <div/>
}
```

**Comment by feross** [comment]
This was discussed here: https://github.com/feross/standard/issues/419 There is a difference in settings.
